src/transform-sdk/cpp/src/transform_sdk.cc

```diff
@@ -464,6 +464,46 @@ on_record_written(const on_record_written_callback& callback) {
     }
 }
 
+namespace sr {
+
+std::expected<std::pair<schema_id, bytes_view>, std::error_code>
+decode_schema_id(bytes_view buf) {
+    constexpr size_t HEADER_LEN = 5U;
+    static const bytes MAGIC_BYTES{0x00};
+    if (
+      buf.subview(0, MAGIC_BYTES.size()) != MAGIC_BYTES
+      || buf.size() < HEADER_LEN) {
+        return std::unexpected{
+          std::make_error_code(std::errc::illegal_byte_sequence)};
+    }
+    // NOLINTBEGIN(*-magic-numbers)
+    auto id_bytes = buf.subview(1, HEADER_LEN - 1);
+    const uint32_t raw_id = (static_cast<uint32_t>(id_bytes[3]) << 0U)
+                            | (static_cast<uint32_t>(id_bytes[2]) << 8U)
+                            | (static_cast<uint32_t>(id_bytes[1]) << 16U)
+                            | (static_cast<uint32_t>(id_bytes[0]) << 24U);
+    // NOLINTEND(*-magic-numbers)
+    if (raw_id > static_cast<uint32_t>(INT_MAX)) {
+        return std::unexpected{
+          std::make_error_code(std::errc::illegal_byte_sequence)};
+    }
+    return std::make_pair(
+      schema_id{static_cast<int32_t>(raw_id)}, buf.subview(HEADER_LEN));
+}
+
+bytes encode_schema_id(schema_id sid, bytes_view buf) {
+    static const bytes MAGIC_BYTES{0x00};
+    bytes be_id;
+    be_id.resize(sizeof(schema_id));
+    std::memcpy(be_id.data(), sid.data(), sizeof(schema_id));
+    bytes result = MAGIC_BYTES;
+    result.append_range(be_id | std::views::reverse);
```

Comment:
There's also `htonl` (https://linux.die.net/man/3/htonl) but this is fine

Reply:
We already require C++23 because of std::expected, so let's use byteswap: https://en.cppreference.com/w/cpp/numeric/byteswap

Reply:
didn't choose `hton` because I think the swap is conditional on host endianness, which shouldn't be an issue in practice but feels like a mild (if extremely common) abuse of the API. might be wrong about that though.

>byteswap

killer. wasn't aware

Reply:
Well if the device is big endian and we receive data in network byte order (which is big endian) hton* is a no-op.

Reply:
yeah that's kinda my point - the endianness is well-defined on both ends (Wasm vs SR header), so what I want is a byteswap, unconditionally. no functional difference in this case though.